### PR TITLE
preProvision machine controller lifecycle hook

### PR DESCRIPTION
**Pull request by rvanderp3**
This PR describes a new Machine controller lifecycle hook that is referred to in https://github.com/openshift/enhancements/pull/1267.  

Associated openshift/api PR: openshift/api#1334

### CODE REVIEW

**Review by JoelSpeed — COMMENTED** [review]
Changes look good, could you do me a favour and retitle/rename the document to reflect that this is now Machine Lifecycle Hooks and not just Machine Deletion Hooks, thanks!

**Comment by damdo** [comment]
> could you do me a favour and retitle/rename the document to reflect that this is now Machine Lifecycle Hooks and not just Machine Deletion Hooks
+1, filename also could be updated, thanks!

File: enhancements/machine-api/machine-deletion-hooks.md [hunk 1]
```diff
@@ -138,6 +139,16 @@ applications first to ensure that service interruptions are minimised in cases
...
 libraries and as such, this would require a custom drain provider.
 
+#### Story 4
+(pre-create) As an operator, I want the ability to utilize my own creation
+controller instead of the logic built into the machine controller. This will
+allow me better flexibility and control over the lifecycle of workloads on each
+node.
+
+For example, this would allow me to configure static IPs for a machine that is 
+about to be created.  IPAM or IP management is not supported today in the
```
Comment (damdo) [inline enhancements/machine-api/machine-deletion-hooks.md:149]:
```suggestion
about to be created. IPAM or IP management is not supported today in the 
```

File: enhancements/machine-api/machine-deletion-hooks.md [hunk 2]
```diff
@@ -223,6 +241,19 @@ application, or manually created and removed by an administrator.
...
+```yaml
+lifecycleHooks:
+  preCreate:
+  - name: <hook-name>
+    owner: <hook-owner>
+```
+
+Hooks defined at this point will prevent the machine-controller from creating 
+the instance in the infrastructure provider and the Machine will not enter the 
+`provisioning` phase until the hooks are removed.
```
Comment (damdo) [inline enhancements/machine-api/machine-deletion-hooks.md:255]: Nit; ref: https://github.com/openshift/api/blob/58db72a40994aaa03e79fd9d7ec791987a5fb1dc/machine/v1beta1/types_machine.go#L311
```suggestion
`Provisioning` phase until the hooks are removed. 
```

**Review by damdo — COMMENTED** [review]
Thanks for updating this, a couple of nits, but LGTM

**Comment by lobziik** [comment]
Aside of file renaming - LGTM. I like this addition.

**Pull request closed without merging**